### PR TITLE
Fix waiting for the TURN server config from the Widget API

### PR DESCRIPTION
**Pull request by weeman1337**
The test changes look more crazy, than they actually are. There is now a test using the deferred TURN server config. It moved into it's own `describe` block for the setup.

The other notable change is the new `initTurnServers` function and calling it when setting up a connection.

**:heavy_check_mark: Checklist**

- [ ] A changeset describing the change and affected packages ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#changelog-and-versioning)).
- [ ] Added or updated documentation.
- [ ] Tests for new functionality and regression tests for bug fixes.
- [ ] Screenshots or videos attached (for UI changes).
- [ ] All your commits have a `Signed-off-by` line in the message ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#dco)).

### CODE REVIEW

File: packages/react-sdk/src/state/communication/webRtcCommunicationChannel.ts [hunk 1]
```diff
@@ -222,6 +197,67 @@ export class WebRtcCommunicationChannel implements CommunicationChannel {
...
+   * If there are already known TURN servers, this is a noop.
+   * Otherwise wait for TURN servers from the Widget API until a timeout.
+   */
+  private async initTurnServers(): Promise<void> {
+    if (this.turnServers !== undefined) {
+      // TURN servers already known
+      return;
+    }
+
+    const source = race(
```
Comment (mgcm) [inline packages/react-sdk/src/state/communication/webRtcCommunicationChannel.ts:210]: seems like race is [being deprecated](https://rxjs.dev/api/operators/race) in rxjs v8?

File: packages/react-sdk/src/state/communication/webRtcCommunicationChannel.ts [hunk 2]
```diff
@@ -222,6 +197,67 @@ export class WebRtcCommunicationChannel implements CommunicationChannel {
...
+  private async handleSessionJoined(session: Session): Promise<void> {
+    const sessionId = this.sessionManager.getSessionId();
+
+    this.logger.log('joined', session.sessionId, session.userId);
+
+    if (!sessionId) {
+      throw new Error('Unknown session id');
+    }
+
+    // Wait for the Widget API or TURN servers to be ready
```
Comment (mgcm) [inline packages/react-sdk/src/state/communication/webRtcCommunicationChannel.ts:232]: wait for the Widget API AND turn servers to be ready?

**Review by mgcm — CHANGES_REQUESTED** [review]

File: packages/react-sdk/src/state/communication/webRtcCommunicationChannel.ts (hunk 1 above)
Comment (weeman1337) [reply packages/react-sdk/src/state/communication/webRtcCommunicationChannel.ts:210]: Argh - I wonder why all the projects are so fancy about breaking their APIs all the time… :neutral_face:

Comment (weeman1337) [reply packages/react-sdk/src/state/communication/webRtcCommunicationChannel.ts]: Done

File: packages/react-sdk/src/state/communication/webRtcCommunicationChannel.ts (hunk 2 above)
Comment (weeman1337) [reply packages/react-sdk/src/state/communication/webRtcCommunicationChannel.ts]: Done

**Review by mgcm — APPROVED** [review]
✅

**Pull request merged**